src/main/java/org/broadinstitute/hellbender/transformers/BaseQualityClipReadTransformer.java

```diff
@@ -29,8 +30,8 @@ public BaseQualityClipReadTransformer(int trim_thresh) {
      *
      */
     @Override
-    public GATKRead apply(GATKRead read) {
-        GATKRead readClippedRightEnd = clipReadRightEnd(read);
+    public GATKRead apply(final GATKRead read) {
+        final GATKRead readClippedRightEnd = clipReadRightEnd(read);
         return clipReadLeftEnd(readClippedRightEnd);
     }
 
```

Comment:
And, while looking at these two functions, they are very similar. 
Considering that even though it is public but seemingly only this class itself is using it, I suggest the following change, but it is totally up to you:

```java
    @override
    public GATKRead apply(final GATKRead read) {
        return clipRead(clipRead(read, false), true);
    }

    private int getClipPoint( final byte[] quals, final boolean fromLeft ) {
        final int readLength = quals.length;
        final int start, end, step;
        if (fromLeft) {
            start = 0;              end = readLength; step = 1;
        } else {
            start = readLength - 1; end = -1;         step = -1;
        }
        int clipSum = 0, lastMax = -1, clipPoint = -1; // -1 means no clip
        for (int i = start; i != end; i+= step) {
            clipSum += (qTrimmingThreshold - quals[i]);
            if (clipSum >= 0 && (clipSum >= lastMax)) {
                lastMax = clipSum;
                clipPoint = i;
            }
        }
        return clipPoint;
    }
    
    private GATKRead clipRead(final GATKRead read, final boolean fromLeft) {
        final byte[] quals = read.getBaseQualities();
        final int clipPoint = getClipPoint(quals, fromLeft);
        if (clipPoint != -1) {
            final ReadClipper readClipper = new ReadClipper(read);
            readClipper.addOp(fromLeft ? new ClippingOp(0, clipPoint) : new ClippingOp(clipPoint, read.getLength()));
            return readClipper.clipRead(ClippingRepresentation.HARDCLIP_BASES);
        } else {
            return read;
        }
    }
    
```